### PR TITLE
feat(sort-handle): hide reorder group title when no sibling groups; disable boundary items instead of hiding

### PR DESCRIPTION
**Pull request by ashetland**
**Related Issue:** #11451

## Summary
Updates calcite-sort-handle to improve its dropdown UX by (1) conditionally omitting the “Reorder” group label when it would be the only visible group and (2) showing boundary reorder actions as disabled items instead of hiding them, along with corresponding Storybook and E2E test updates.

**Changes:**

- Hide the reorder dropdown-group title when no Move/Add sibling groups are present.
- Render all reorder actions consistently and disable boundary actions (top/up/down/bottom) instead of conditionally not rendering them.
- Update Storybook stories and add/adjust E2E coverage for the new behaviors.

### CODE REVIEW

File: packages/components/src/components/sort-handle/sort-handle.tsx [hunk 1]
```diff
@@ -47,15 +47,29 @@ export class SortHandle extends LitElement {
...
+    return this.moveToItems.length > 0;
+  }
+
+  get hasAddToItems(): boolean {
+    return this.addToItems.length > 0;
+  }
+
+  get reorderGroupTitle(): string {
+    return this.hasMoveToItems || this.hasAddToItems ? this.messages.reorder : null;
+  }
```
[the remark this answers is not in the document]
Comment (ashetland) [reply packages/components/src/components/sort-handle/sort-handle.tsx:69]: I think we can ignore this since it would add functionality that doesn't exist currently.

File: packages/components/src/components/sort-handle/sort-handle.tsx [hunk 2]
```diff
@@ -47,15 +47,29 @@ export class SortHandle extends LitElement {
...
+  get hasMoveToItems(): boolean {
+    return this.moveToItems.length > 0;
+  }
+
+  get hasAddToItems(): boolean {
+    return this.addToItems.length > 0;
+  }
+
+  get reorderGroupTitle(): string {
+    return this.hasMoveToItems || this.hasAddToItems ? this.messages.reorder : null;
```
[the remark this answers is not in the document]
Comment (ashetland) [reply packages/components/src/components/sort-handle/sort-handle.tsx]: @driskull Thoughts?

File: packages/components/src/components/sort-handle/sort-handle.tsx [hunk 3]
```diff
@@ -412,30 +427,32 @@ export class SortHandle extends LitElement {
...
 
-    return setPosition !== setSize && setPosition !== setSize - 1
-      ? this.renderDropdownItem(3, this.messages.moveToBottom)
-      : null;
+    return this.renderDropdownItem(
+      3,
+      this.messages.moveToBottom,
+      setPosition === setSize || setPosition === setSize - 1,
+    );
   }
```
[the remark this answers is not in the document]
Comment (ashetland) [reply packages/components/src/components/sort-handle/sort-handle.tsx:484]: @driskull Thoughts on this too?

File: packages/components/src/components/sort-handle/sort-handle.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (driskull) [reply packages/components/src/components/sort-handle/sort-handle.tsx]: @ashetland I think this should return an empty string instead of undefined.

In the future, if a11y is a concern, we may need to add a `label` property to `dropdown-group` in order to still cover a11y since we are removing the display of `Reorder` and just showing the items ` without a `groupTitle` for the group.

File: packages/components/src/components/sort-handle/sort-handle.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (driskull) [reply packages/components/src/components/sort-handle/sort-handle.tsx:484]: > When neither setPosition nor setSize is provided,

@ashetland They should be provided so this is really an edge case. These changes are fine, however they may be out of scope for this issue and would likely need some tests added.

IMO this could be a followup issue or ignored.

[the remark this answers is not in the document]
Comment (ashetland) [reply packages/components/src/components/sort-handle/sort-handle.tsx:484]: I think we can ignore for now.

File: packages/components/src/components/sort-handle/sort-handle.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (ashetland) [reply packages/components/src/components/sort-handle/sort-handle.tsx]: Updated to return an empty string.

**Comment by ashetland** [comment]
@driskull @jcfranco Can you take another look at this?

File: packages/components/src/components/sort-handle/sort-handle.browser.e2e.tsx [hunk 4]
```diff
@@ -70,3 +76,81 @@ describe("translation support", () => {
...
   disabled(() => mount(<calcite-sort-handle label="test" set-position="4" set-size="10" />));
 });
+
+it("renders disabled boundary reorder items instead of hiding them", async () => {
+  const { el, reRender } = await mount(
+    <calcite-sort-handle label="test" set-position="1" set-size="4" />,
+  );
+  await reRender();
+
+  const reorderGroup = el.shadowRoot.querySelector<HTMLElement>(`#${IDS.reorder}`);
```
Comment (driskull) [inline packages/components/src/components/sort-handle/sort-handle.browser.e2e.tsx]: Can you prompt copilot to use `vite selectors` instead of `querySelector`? That should change these from querying the shadow DOM.

File: packages/components/src/components/sort-handle/sort-handle.browser.e2e.tsx [hunk 5]
```diff
@@ -10,6 +10,12 @@ import {
   renders,
   t9n,
 } from "../../tests/commonTests/browser";
+import { IDS } from "./resources";
+import { SortHandle } from "./sort-handle";
+
+type DropdownLike = HTMLElement & { disabled: boolean };
```
Comment (driskull) [inline packages/components/src/components/sort-handle/sort-handle.browser.e2e.tsx]: Instead of using these `ComponentLike` typings, can you import the `Dropdown` file and use `Dropdown["el"]`?

File: packages/components/src/components/sort-handle/sort-handle.tsx [hunk 6]
```diff
@@ -268,6 +301,26 @@ export class SortHandle extends LitElement {
     this.calciteSortHandleAdd.emit({ addTo });
   }
 
+  private isTopReorderDisabled(): boolean {
```
Comment (driskull) [inline packages/components/src/components/sort-handle/sort-handle.tsx]: It may be nicer if these functions were getter properties instead.

**Review by driskull — CHANGES_REQUESTED** [review]
Code changes look good. Just some cleanup of tests and maybe changing private methods to private getter properties

File: packages/components/src/components/sort-handle/sort-handle.browser.e2e.tsx (hunk 4 above)
Comment (ashetland) [reply packages/components/src/components/sort-handle/sort-handle.browser.e2e.tsx]: Updated

File: packages/components/src/components/sort-handle/sort-handle.tsx (hunk 6 above)
Comment (ashetland) [reply packages/components/src/components/sort-handle/sort-handle.tsx]: Updated

**Review by driskull — APPROVED** [review]
LGTM!

File: packages/components/src/components/sort-handle/sort-handle.tsx [hunk 7]
```diff
@@ -47,15 +47,29 @@ export class SortHandle extends LitElement {
...
 
+  get hasMoveToItems(): boolean {
+    return this.moveToItems.length > 0;
+  }
+
+  get hasAddToItems(): boolean {
+    return this.addToItems.length > 0;
+  }
+
+  get reorderGroupTitle(): string {
```
Comment (eriklharper) [inline packages/components/src/components/sort-handle/sort-handle.tsx:67]: Feels like this should be named `getGroupTitle` instead since it doesn't actually do any reordering.

File: packages/components/src/components/sort-handle/sort-handle.stories.ts [hunk 8]
```diff
@@ -49,27 +49,43 @@ export const positions = (): string => html`
...
-    const sortHandle = document.querySelector("calcite-sort-handle");
-    sortHandle.addToItems = [
-      { element: document.createElement("div"), id: "1", label: "Group 1" },
-      { element: document.createElement("div"), id: "2", label: "Group 2" },
-    ];
+    (async () => {
+      await customElements.whenDefined("calcite-sort-handle");
+
+      const sortHandle = document.querySelector("#add-to-story-handle");
+      await sortHandle.componentOnReady();
```
Comment (eriklharper) [inline packages/components/src/components/sort-handle/sort-handle.stories.ts:80]: are these `await`s actually necessary?

File: packages/components/src/components/sort-handle/sort-handle.tsx (hunk 7 above)
Comment (driskull) [reply packages/components/src/components/sort-handle/sort-handle.tsx:67]: reoder is the group name, but I get how that could be confusing :)

Comment (eriklharper) [reply packages/components/src/components/sort-handle/sort-handle.tsx:67]: Gotcha, yeah, this is totally a minor thing, not a big deal, thanks for the context!
